src/User.js

```diff
@@ -14,37 +14,45 @@ class User {
     return firstName[0];
   }
 
-  calculateAvgOunces(hydrationData) {
-    const currentUser = hydrationData.filter(element => {
+  findUser(dataset) {
+    return dataset.filter(element => {
       return element.userID === this.id
-    })
+    }) 
+  }
+
+  findUserAndDate(dataset, date) {
```

Comment:
Although there is overlap in the functionality of findUser and findUserAndDate, I agree that this approach is clearer than using findUser within findUserAndDate.